sklearn/cluster/_kmeans.py

```diff
@@ -990,7 +1003,12 @@ def _init_centroids(
                 x_squared_norms=x_squared_norms,
             )
         elif isinstance(init, str) and init == "random":
-            seeds = random_state.permutation(n_samples)[:n_clusters]
+            seeds = random_state.choice(
+                n_samples,
+                size=n_clusters,
+                replace=False,
+                p=sample_weight / sample_weight.sum(),
+            )
```

Comment:
Note that this is a backward incompatible change, even without sample weights because it won't give the same result as before for the same random_state. It will require an entry in the changed models section of the what's new

Reply:
Done